sklearn/mixture/tests/test_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,264 @@
+    _test_inputs(X, ValueError, linkage="1")
+
+    # covariance type is not an array, string or list
+    _test_inputs(X, TypeError, covariance_type=1)
+
+    # covariance type is not in ['spherical', 'diag', 'tied', 'full']
+    _test_inputs(X, ValueError, covariance_type="1")
+
+    # euclidean is not an affinity option when ward is a linkage option
+    _test_inputs(X, ValueError, **{"affinity": "manhattan", "linkage": "ward"})
+
+    # criterion is not "aic" or "bic"
+    _test_inputs(X, ValueError, criterion="cic")
+
+
+def test_labels_init():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    # label_init is not a 1-D array
+    _test_inputs(X, TypeError, label_init=np.zeros([100, 2]))
+
+    # label_init is not 1-D array, a list or None.
+    _test_inputs(X, TypeError, label_init="label")
+
+    # label_init length is not equal to n_samples
+    _test_inputs(X, ValueError, label_init=np.zeros([50, 1]))
+
+    # label_init length does not match min_components and max_components
+    _test_inputs(
+        X,
+        ValueError,
+        **{
+            "label_init": np.zeros([100, 1]),
+            "min_components": 2,
+            "max_components": 3,
+        },
+    )
+
+
+def test_predict_without_fit():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    with pytest.raises(NotFittedError):
+        gmIC = GaussianMixtureIC(min_components=2)
+        gmIC.predict(X)
+
+
+def test_cosine_with_0():
+    X = np.array(
+        [
+            [0, 1, 0],
+            [1, 0, 1],
+            [0, 0, 0],
+            [1, 1, 0],
+            [0, 0, 1],
+            [0, 1, 1],
+            [1, 1, 1],
+            [1, 0, 0],
+            [0, 1, 1],
+            [1, 1, 0],
+            [0, 1, 0],
+        ]
+    )
+
+    with pytest.warns(UserWarning):
+        gmIC = GaussianMixtureIC(min_components=2, affinity="all")
+        gmIC.fit(X)
+
+
+def _test_two_class(**kws):
+    """
+    Easily separable two gaussian problem.
+    """
+    np.random.seed(1)
+
+    n = 100
+    d = 3
+
+    X1 = np.random.normal(2, 0.5, size=(n, d))
+    X2 = np.random.normal(-2, 0.5, size=(n, d))
+    X = np.vstack((X1, X2))
+    y = np.repeat([0, 1], n)
+
+    # test BIC
+    gmIC = GaussianMixtureIC(max_components=5, criterion="bic", **kws)
+    gmIC.fit(X, y)
+    n_components = gmIC.n_components_
+
+    # Assert that the two cluster model is the best
+    assert_equal(n_components, 2)
+
+    # Asser that we get perfect clustering
+    ari = adjusted_rand_score(y, gmIC.fit_predict(X))
+    assert_allclose(ari, 1)
+
+    # test AIC
+    gmIC = GaussianMixtureIC(max_components=5, criterion="aic", **kws)
+    gmIC.fit(X, y)
+    n_components = gmIC.n_components_
+
+    # AIC gets the number of components wrong
+    assert_equal(n_components >= 1, True)
+    assert_equal(n_components <= 5, True)
+
+
+def test_two_class_parallel():
+    _test_two_class(n_jobs=2)
```

Comment:
Maybe integrate the content in `_test_two_class` in `test_two_class_parallel` directly?

Reply:
so I guess I should just specify `n_jobs=2` in `test_two_class` so both the 2-class Gaussian problem and parallel processing are tested?